src/ts/extension.ts

```diff
@@ -114,56 +141,55 @@ export async function activate(context: vscode.ExtensionContext) {
   );
 
   context.subscriptions.push(
-    vscode.commands.registerCommand("pkl.downloadPackage", async (packageUri: string) => {
-      await client.sendRequest(pklDownloadPackageRequest, packageUri);
+    vscode.commands.registerCommand(COMMAND_DOWNLOAD_PACKAGE, async (packageUri: string) => {
+      if (languageClientRef.client === undefined) {
+        return;
+      }
+      await languageClientRef.client.sendRequest(pklDownloadPackageRequest, packageUri);
     })
   );
 
   context.subscriptions.push(
-    vscode.commands.registerCommand("pkl.syncProjects", async () => {
-      await client.sendRequest(pklSyncProjectsRequest, null);
+    vscode.commands.registerCommand(COMMAND_SYNC_PROJECTS, async () => {
+      if (languageClientRef.client === undefined) {
+        return;
+      }
+      await languageClientRef.client.sendRequest(pklSyncProjectsRequest, null);
     })
   );
 
   context.subscriptions.push(
-    vscode.commands.registerCommand("pkl.configure", async (configurationPath: string) => {
-      await vscode.commands.executeCommand("workbench.action.openSettings", configurationPath);
+    vscode.commands.registerCommand(COMMAND_PKL_CONFIGURE, async (configurationPath: string) => {
+      await vscode.commands.executeCommand(COMMAND_OPEN_WORKSPACE_SETTINGS, configurationPath);
     })
   );
-
-  registerNotificationHandlers(client);
 }
 
-// this method is called when your extension is deactivated
-export function deactivate(): Thenable<void> | undefined {
-  if (!client) return undefined;
-  return client.stop();
+async function askForLspJar() {
+  const response = await vscode.window.showWarningMessage(
+    "Path to pkl-lsp.jar not configured",
+    "Configure path to pkl-lsp.jar"
+  );
+  if (response === "Configure path to pkl-lsp.jar") {
```

Comment:
Shouldn't this be a `const.ts` member also?

Reply:
I'll move this to its own const, but I don't think this one belongs in `consts.ts`. It might belong in a messages bundle type thing though.